src/stravalib/model.py

```diff
@@ -220,6 +231,148 @@ def check_valid_location(
         return None
 
 
+# Custom types:
+
+BaseType = TypeVar("BaseType")
+CustomType = TypeVar("CustomType")
+
+
+class _CustomTypeAnnotation(Generic[BaseType, CustomType], ABC):
+    _type: Type[CustomType]
+
+    @classmethod
+    @abstractmethod
+    def validate(cls, value: BaseType) -> CustomType: ...
+
+    @classmethod
+    @abstractmethod
+    def get_core_schema(cls) -> Mapping[str, Any]: ...
+
+    @classmethod
+    def __get_pydantic_core_schema__(
+        cls,
+        _source_type: BaseType,
+        _handler: GetCoreSchemaHandler,
+    ) -> core_schema.CoreSchema:
+        from_schema = core_schema.chain_schema(
+            [
+                cls.get_core_schema(),
+                core_schema.no_info_plain_validator_function(cls.validate),
+            ]
+        )
+
+        return core_schema.json_or_python_schema(
+            json_schema=from_schema,
+            python_schema=core_schema.union_schema(
+                [core_schema.is_instance_schema(cls._type), from_schema]
+            ),
+            serialization=core_schema.plain_serializer_function_ser_schema(
+                lambda v: v
+            ),
+        )
+
+    @classmethod
+    def __get_pydantic_json_schema__(
+        cls,
+        _core_schema: core_schema.CoreSchema,
+        handler: GetJsonSchemaHandler,
+    ) -> JsonSchemaValue:
+        return handler(cls.get_core_schema())
+
+
+class _CustomIntAnnotation(_CustomTypeAnnotation[int, Any], ABC):
+    @classmethod
+    def get_core_schema(cls) -> Mapping[str, Any]:
+        return core_schema.int_schema()
+
+
+class _CustomFloatAnnotation(_CustomTypeAnnotation[float, Any], ABC):
+    @classmethod
+    def get_core_schema(cls) -> Mapping[str, Any]:
+        return core_schema.float_schema()
+
+
+class _CustomStrAnnotation(_CustomTypeAnnotation[str, Any], ABC):
+    @classmethod
+    def get_core_schema(cls) -> Mapping[str, Any]:
+        return core_schema.str_schema()
+
+
+class Duration(int):
```

Comment:
The suggestion is good; I like these examples! Regarding your question: the example should be

```python
activity.elapsed_time.timedelta()
```

as a method call. This will give you the timedelta object (which can be formatted with `str()`).